bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java

```diff
@@ -106,32 +148,33 @@ void changeChannel(String channel, State state) {
 
     private void processIncomingState(String messageJSON) {
         // Need to handle State and Level at the same time to process level=0 as off//
-        BigDecimal tempBulbLevel = BigDecimal.ZERO;
+        PercentType tempBulbLevel = PercentType.ZERO;
         String bulbState = Helper.resolveJSON(messageJSON, "\"state\":\"", 3);
         String bulbLevel = Helper.resolveJSON(messageJSON, "\"level\":", 3);
         if (!bulbLevel.isEmpty()) {
             if ("0".equals(bulbLevel) || "OFF".equals(bulbState)) {
                 changeChannel(CHANNEL_LEVEL, OnOffType.OFF);
-                tempBulbLevel = BigDecimal.ZERO;
```

Comment:
Should this be changed to the following instead of being removed?
tempBulbLevel = PercentType.ZERO;
I do not remember the reason as it has been awhile, but it would have been there for a reason. If you feel it should be removed I would need to do some tests to look for a bug.

Reply:
It will already be set to PercentType.ZERO as that's what it's initialized to on line 151.

Reply:
Sorry my bad, closing comment.